### PR TITLE
Ambiguous #36

### PR DESCRIPTION
**Pull request by Galileo1**
@gkushang ,

Here, is the PR to support the "ambiguous" status for cucumber steps.
I have made some significant changes as per my knowledge and as a consumer of the report but please free to discuss if these changes doesn't make sense.

1. Changes for ambiguous status.
2. Changed the "Undefined" to show "label-default" to distinguish it from "Pending"  ("label-info').
3. As per my understanding we don't need to display the "Skipped" scenario because a step will be skipped if the one of the previous step has failed/is undefined/is pending.So the report should show actual status in scenario pie chart.
image

image

image

4. I changed this block "_.drop(element.tags, feature.tags.length)" because if features doesn't have tags and scenario does then it will break.
5. Fixed and changed the "foundation.tmpl" to show the actual status.
image
image

Please review the PR and let me know what you think.

thanks

### CODE REVIEW

**Comment by Galileo1** [comment]
@gkushang did it failed because it doesn't removes the files after creating them??

**Comment by gkushang** [comment]
the CI failed due to ``` Multiple step definitions match:
       /^cucumber-html-reporter should create HTML report/                       - test/features/step_definitions/stepDefs.js:60
       /^cucumber-html-reporter should create HTML report with ambiguous steps$/ - test/features/step_definitions/stepDefs.js:74
```
https://travis-ci.org/gkushang/cucumber-html-reporter/builds/178146903

**Comment by Galileo1** [comment]
@gkushang Yeah that's expected because it ambiguous.  So does that mean that I shouldn't be having any test that failing?And I shud submit it as passing test even though the test is about cucumber failing? I am just correlating it because at the moment one of failing scenario- "Fred wants to see a screenshot attached to HTML report" is submitted as passing it doesn't fail. I need to make changes to so that it fails during testing.

**Comment by gkushang** [comment]
@Galileo1 yeah, that is something needs to be figured out. Since `npm test` runs cucumber, it fails the build due to some of the valid feature that we wanted to test as fail. Until we figure that out, lets submit only passing scenarios. Sounds good?

**Comment by Galileo1** [comment]
@gkushang Thanks and done.Can you please review the PR.

File: templates/foundation/features.tmpl [hunk 1]
```diff
@@ -30,12 +33,16 @@
...
                   <span class="alert label right">Failed</span>
+                <% } else if(step.result.status === 'ambiguous') { %>
+                  <span class="label right">Ambiguous</span>
                 <% } else if(step.result.status === 'skipped') { %>
                   <span class="label right">Skipped</span>
-                <% } else if(step.result.status === 'undefined' || step.result.status === 'pending') { %>
+                <% } else if(step.result.status === 'undefined') { %>
                   <span class="label right">undefined</span>
+                <% } else if(step.result.status === 'pending') { %>
+                  <span class="label right">pending</span>
```
Comment (gkushang) [inline templates/foundation/features.tmpl]: `pending` should be `Pending`

**Comment by Galileo1** [comment]
@gkushang - refactored after review comments.Can I please have another review for PR.

**Pull request merged**